validator/src/pylogger.rs

```diff
@@ -42,12 +42,12 @@ pub fn set_up_logger(verbosity: u64, py: Python) {
 
 #[no_mangle]
 #[allow(unused)]
-#[allow(private_no_mangle_fns)]
 pub extern "C" fn pylogger_init(verbosity: usize) {
     let gil = Python::acquire_gil();
     let py = gil.python();
-    PyLogger::init(determine_log_level(verbosity as u64), py)
-        .expect("Failed to initialize PyLogger");
+    if let Err(_) = PyLogger::init(determine_log_level(verbosity as u64), py) {
+        warn!("Attempted to initialize logger twice; ignoring");
```

Comment:
Is that the only way PyLogger::init() can fail? If not, should the error text be logged?

Reply:
Yes, it only returns the `SetLoggerError` which is only returned if a logger has been set twice.

Reply:
See: https://docs.rs/log/0.4.6/log/struct.SetLoggerError.html